src/App.tsx

```diff
@@ -1,26 +1,330 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  updateTodo,
+  USER_ID,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotification } from './components/ErrorNotification';
+import { ErrorType, FilterType } from './types/ErrorType';
+import cn from 'classnames';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { TodoItem } from './components/TodoItem';
+import { StatusFilter } from './components/StatusFilter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorType>(
+    ErrorType.DEFAULT,
+  );
+  const [isLoading, setIsLoading] = useState(false);
+  const [status, setStatus] = useState<FilterType>(FilterType.All);
+  const [processingTodoIds, setProcessingTodoIds] = useState<Todo['id'][]>([]);
+  const [title, setTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [editingTodoIds, setEditingTodoIds] = useState<number[]>([]);
+
+  const handleAddTodoToProcessing = (todoId: Todo['id']) => {
+    setProcessingTodoIds(current => [...current, todoId]);
+  };
+
+  const handleRemoveTodoToProcessing = (todoId: Todo['id']) => {
+    setProcessingTodoIds(current => current.filter(id => id !== todoId));
+  };
+
+  const handleAddTodoToEditing = (todoId: number) => {
+    setEditingTodoIds(current => [...current, todoId]);
+  };
+
+  const handleRemoveTodoToEditing = (todoId: number) => {
+    setEditingTodoIds(current => current.filter(id => id !== todoId));
+  };
```

Comment:
you have state with ids and than ypu pass it as loading in TodoItem
use this info instead of extra states and handlers